app/observers/live_events_observer.rb

```diff
@@ -19,9 +20,19 @@ def after_update(obj)
       if obj.title_changed? || obj.body_changed?
         Canvas::LiveEvents.wiki_page_updated(obj, obj.title_changed? ? obj.title_was : nil,
                                                   obj.body_changed? ? obj.body_was : nil)
+      elsif obj.workflow_state_changed? && obj.workflow_state == 'deleted'
+        # Wiki pages are soft deleted rather than destroyed
+        Canvas::LiveEvents.wiki_page_deleted(obj)
       end
     when Assignment
       Canvas::LiveEvents.assignment_updated(obj)
+    when Attachment
+      if obj.display_name_changed?
+        Canvas::LiveEvents.attachment_updated(obj, obj.display_name_was)
+      elsif obj.file_state_changed? && obj.file_state == 'deleted'
+        # Attachments are soft deleted rather than destroyed
```

Comment:
Nearly always AFAICT, but there were a few spots where `destroy_permanently` is called, so I added the `after_destroy` trigger back in.